.travis.yml

```diff
@@ -4,6 +4,9 @@ node_js:
   - '8'
   - '6'
   - '4'
+matrix:
+  allow_failures:
+  - node_js: '4'
```

Comment:
Does it fail on node 4? Not a fan of allowing failures for it, or we need to remove it from the supported version (major version bump).

Reply:
Yes it does, but unrelated to this PR (master is already failing).

I didn't want to remove it in this PR to keep it focused on the TS changes.

Reply:
I see, only the linter fails, please remove this from this PR and we're going to patch this in another one thanks!